Floofbot/Services/EventLoggerService.cs

```diff
@@ -0,0 +1,562 @@
+                    .AddField("Joined Discord", user.CreatedAt, true)
+                    .WithFooter($"user_join user_joinlog {user.Id}")
+                    .WithCurrentTimestamp();
+
+                    if (Uri.IsWellFormedUriString(user.GetAvatarUrl(), UriKind.Absolute))
+                        embed.WithThumbnailUrl(user.GetAvatarUrl());
+                    await channel.SendMessageAsync("", false, embed.Build());
+                }
+                catch (Exception ex)
+                {
+                    Log.Error("Error with the user joined event handler: " + ex);
+                    return;
+                }
+            });
+            return Task.CompletedTask;
+        }
+        public Task UserLeft(IGuildUser user)
+        {
+            var _ = Task.Run(async () =>
+            {
+                try
+                {
+                    if ((IsToggled(user.Guild)) == false)
+                        return;
+
+                    Discord.ITextChannel channel = await GetChannel(user.Guild, "UserLeftChannel");
+                    if (channel == null)
+                        return;
+
+                    var embed = new EmbedBuilder()
+                    .WithTitle($"❌ User Left | {user.Username}#{user.Discriminator}")
+                    .WithColor(Color.Red)
+                    .WithDescription($"{user.Mention} | ``{user.Id}``")
+                    .WithFooter($"user_leave user_joinlog {user.Id}")
+                    .WithCurrentTimestamp();
+
+                    if (Uri.IsWellFormedUriString(user.GetAvatarUrl(), UriKind.Absolute))
+                        embed.WithThumbnailUrl(user.GetAvatarUrl());
+
+                    await channel.SendMessageAsync("", false, embed.Build());
+                }
+                catch (Exception ex)
+                {
+                    Log.Error("Error with the user left event handler: " + ex);
+                    return;
+                }
+            });
+            return Task.CompletedTask;
+
+        }
+        public Task GuildMemberUpdated(SocketGuildUser before, SocketGuildUser after)
+        {
+            var _ = Task.Run(async () =>
+            {
+                try
+                {
+                    if (before == null || after == null) // empty user params
+                        return;
+                    var user = after as SocketGuildUser;
+
+                    if ((IsToggled(user.Guild) == false)) // turned off
+                        return;
+
+                    Discord.ITextChannel channel = await GetChannel(user.Guild, "MemberUpdatesChannel");
+                    if (channel == null) // no log channel set
+                        return;
+
+                    var embed = new EmbedBuilder();
+
+                    if (before.Username != after.Username)
+                    {
+                        embed.WithTitle($"👥 Username Changed | {user.Username}#{user.Discriminator}")
+                            .WithColor(Color.Purple)
```

Comment:
General note on style: definitely not a PR blocker, but I'd like to point out that we should probably stick to and/or agree on a certain code style. The only thing here is that the continuation indentation is inconsistent with some others in this file.

Reply:
I will fix this - wasnt intentional!